### PR TITLE
Add CircleCI

### PR DESCRIPTION
**Pull request by rodrigok**
@RocketChat/core 

After artifact is uploaded to s3 a redirect is put in place for the following:
* downloads.rocket.chat/stable
* downloads.rocket.chat/candidate
* downloads.rocket.chat/develop

Also develop releases a redirect is added for:
* downloads.rocket.chat/builds/rocket.chat-${RC_VERSION}-develop.tgz which redirects to the latest develop build

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+          source .circleci/setartname.sh
+          source .circleci/setdeploydir.sh
+          bash .circleci/setupsig.sh
+          bash .circleci/namefiles.sh
+          # echo ".circleci/sandstorm.sh"
+
+          aws s3 sync $ROCKET_DEPLOY_DIR/ s3://download.rocket.chat/
+
+          bash .circleci/docker.sh
+          bash .circleci/update-releases.sh
```
Comment (geekgonecrazy) [inline .circleci/config.yml]: This and corresponding file I assume we no longer need, as that endpoint no longer exists.

Comment (sampaiodiego) [reply .circleci/config.yml]: oh, good to know :+1:

Comment (geekgonecrazy) [reply .circleci/config.yml]: Disregard removing.  Just updated the contents of the file :)

**Comment by geekgonecrazy** [comment]
Updated Description of PR with a bit more info from my changes:

---

After artifact is uploaded to s3 a redirect is put in place for the following:
* downloads.rocket.chat/stable
* downloads.rocket.chat/candidate
* downloads.rocket.chat/develop

Also develop releases a redirect is added for:
* downloads.rocket.chat/builds/rocket.chat-${RC_VERSION}-develop.tgz which redirects to the latest develop build

**Pull request merged**